helix-term/src/ui/editor.rs

```diff
@@ -697,6 +738,196 @@ impl EditorView {
+            .screen_coords_at_pos(doc, text, cursor_line)
+            .unwrap_or_default()
+            .row as u16;
+
+        if visual_cursor_pos == 0 {
+            return None;
+        }
+
+        if let Some(nodes) = nodes {
+            if nodes.iter().any(|node| {
+                node.top_first_byte == top_first_byte && (visual_cursor_pos as usize) >= nodes.len()
+            }) {
+                return Some(nodes.to_vec());
+            }
+        }
+
+        let context_nodes = doc
+            .language_config()
+            .and_then(|lc| lc.sticky_context_nodes.as_ref());
+
+        let mut parent = tree
+            .root_node()
+            .descendant_for_byte_range(top_first_byte, top_first_byte)
+            .and_then(|n| n.parent());
+
+        // context is list of numbers of lines that should be rendered in the LSP context
+        let mut context: Vec<StickyNode> = Vec::new();
+
+        while let Some(node) = parent {
+            let line = text.char_to_line(node.start_byte());
+
+            // if parent of previous node is still on the same line, use the parent node
+            if let Some(prev_line) = context.last() {
+                if prev_line.line_nr == line {
+                    context.pop();
+                }
+            }
+
+            if context_nodes.map_or(true, |nodes| nodes.iter().any(|n| n == node.kind())) {
+                context.push(StickyNode {
+                    visual_line: 0, // with sorting it will be done
+                    line_nr: line,
+                    indicator: None,
+                    top_first_byte,
+                });
+            }
+
+            parent = node.parent();
+        }
+
+        // context should be filled by now
+        if context.is_empty() {
+            return None;
+        }
+
+        let max_lines = config.sticky_context.max_lines;
+        let max_nodes_amount = if max_lines == 0 {
+            viewport.height as usize / 3
+        } else {
+            max_lines.min(viewport.height) as usize
+        };
+
+        // we render from top most (last in the list)
+        context = context
+            .into_iter()
+            // only take the nodes until 1 / 3 of the viewport is reached or the maximum amount of sticky nodes
+            .take(max_nodes_amount)
+            .rev()
+            .enumerate()
+            .take_while(|(i, _)| *i + 1 != visual_cursor_pos as usize) // also only nodes that don't overlap with the visual cursor position
+            .map(|(i, node)| {
+                let mut new_node = node;
+                new_node.visual_line = i as u16;
+                new_node
+            })
+            .collect();
+
+        if config.sticky_context.indicator {
+            let mut str = String::new();
+            let message = "┤Sticky Context├";
+            let side_placeholder = (viewport.width as usize)
+                .saturating_div(2)
+                .saturating_sub(message.len() - 1);
```

Comment:
you're using Unicode chars in the message and then computing `message.len() - 1` here which works in bytes, thats probably not the correct action